### ISSUE TITLE
The Typescript definitions aren't up to date

### ISSUE THREAD

**Issue by JanBaryla** [alexbol99/flatten-js#45]
If you look closely, some definitions are off.

For example - in the definitions, it's allowed to pass an instance of Box to the constructor of a Polygon. The documentation doesn't mention it and it doesn't work.

Another instance are the Box `toSegments` and `toPoints` methods. They aren't present in the type definitions at all.

**Comment by adamdva** [comment]
Using Typescript, I don't seam to be able to use Flatten.Relations
Relations is undefined so Flatten.Relations.intersect(box, shape) results in:
ERROR: TypeError: Cannot read property 'intersect' of undefined

**Comment by alexbol99** [comment]
Hello, @JanBaryla 
- I fixed definitions you mentioned and some others in .d.ts file. Please send me more if you find.
- As for polygon construction with box, I think it works, at least in my tests:
```javascript
let polygon = new Polygon(box(30,40,100,200));
expect(polygon.faces.size).to.equal(1);
expect(polygon.edges.size).to.equal(4);
expect([...polygon.faces][0].orientation()).to.equal(Flatten.ORIENTATION.CCW); 
```
Thank,
Alex

**Comment by alexbol99** [comment]
Hello, @adamdva 
Issue with exporting namespaces `BooleanOperations` and `Relations` fixed in the patch v1.2.8
Thanks,
Alex
